src/bin/cargo/cli.rs

```diff
@@ -382,19 +399,55 @@ fn config_configure(
     Ok(())
 }
 
+/// Precedence isn't the most obvious from this function because
+/// - Some is determined by `expand_aliases`
+/// - Some is enforced by `avoid_ambiguity_between_builtins_and_manifest_commands`
+///
+/// In actuality, it is:
+/// 1. built-ins xor manifest-command
+/// 2. aliases
+/// 3. external subcommands
 fn execute_subcommand(config: &mut Config, cmd: &str, subcommand_args: &ArgMatches) -> CliResult {
     if let Some(exec) = commands::builtin_exec(cmd) {
         return exec(config, subcommand_args);
     }
 
-    let mut ext_args: Vec<&OsStr> = vec![OsStr::new(cmd)];
-    ext_args.extend(
-        subcommand_args
-            .get_many::<OsString>("")
-            .unwrap_or_default()
-            .map(OsString::as_os_str),
-    );
-    super::execute_external_subcommand(config, cmd, &ext_args)
+    if commands::run::is_manifest_command(cmd) {
```

Comment:
That is a possibility.  Another one is to move more of the external subcommand contract into `execute_external_subcommand` (ie have it do the merging).

I went ahead and duplicated because I was optimizing for a clean deletion when this is stabililized (or at least the we stop warning)

Reply:
Oh, I was thinking bigger scope than your comment.  I guess?  Doesn't seem like much gain but either way

Reply:
This new variable would have to live aside `ext_args` and have a name clear enough to get get confusing.  Personally, I don't think this change is worth it.